.github/workflows/ci.yml

```diff
@@ -7,99 +7,81 @@ jobs:
         runs-on: ubuntu-latest
         strategy:
             matrix:
-                python-version: [3.6, 3.7, 3.8, 3.9]
+                python-version: ['3.9', '3.10', '3.11', '3.12']
 
         steps:
-            - uses: actions/checkout@v1
+            - uses: actions/checkout@v4
             - name: Set up Python ${{ matrix.python-version }}
-              uses: actions/setup-python@v2
+              uses: actions/setup-python@v5
               with:
                   python-version: ${{ matrix.python-version }}
             - name: Install dependencies
               run: |
                   python -m pip install --upgrade pip
                   pip install -U -r requirements_dev.txt
                   pip install tox tox-gh-actions
-            - name: Test with tox
+            - name: Test with tox numpy>=2
+              run: tox
+            - name: Install numpy < 2
+              run: |
+                  pip uninstall numpy
+                  pip install 'numpy<2'
+            - name: Test with tox numpy<2
               run: tox
 
     build_wheels:
-        name: Build wheel for cp${{ matrix.python }}-${{ matrix.platform_id }}-${{ matrix.manylinux_image }}
+        name: Build wheel for cp${{ matrix.python }}-${{ matrix.arch }}
         runs-on: ${{ matrix.os }}
         strategy:
             # Ensure that a wheel builder finishes even if another fails
             fail-fast: false
             matrix:
                 os: [windows-latest, ubuntu-latest, macos-latest]
-                python: [36, 37, 38, 39]
-                bitness: [32, 64]
-                manylinux_image: [manylinux1, manylinux2010]
+                python: [38, 39, 310, 311, 312]
+                arch: [auto64]
```

Comment:
Do we need to do some thinking to ensure that wheels are built for macos x86?

Looks like we should be able to cross compile, but might mean setting `universal2` as the arch for the macos job. Just skimmed:
- https://cibuildwheel.pypa.io/en/stable/options/#archs
- https://cibuildwheel.pypa.io/en/stable/faq/#apple-silicon